src/main/java/core/mate/academy/model/Bulldozer.java

```diff
@@ -1,15 +1,34 @@
 package core.mate.academy.model;
 
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Add some custom fields that could be only in Bulldozer
  * Do not remove no-args constructor
  */
-public class Bulldozer extends Machine {
+public class Bulldozer extends Machine implements MachineProducer<Bulldozer> {
```

Comment:
separate producers from the models
You have three types of machine, and you should have three separate producers